### PR TITLE
Add KoLeoLoss

### PR DESCRIPTION
**Pull request by guarin**
## Changes

* Add KoLeoLoss

## How was it tested?

* Added unit test

### CODE REVIEW

File: tests/loss/test_koleo_loss.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+class TestKoLeoLoss:
+    # Test values generated using the original implementation from:
+    # https://github.com/facebookresearch/dinov2/blob/main/dinov2/loss/koleo_loss.py
+    @pytest.mark.parametrize(
+        "x, expected_loss",
+        [
+            (torch.tensor([[1.0]]), 17.7275),
+            (torch.tensor([[1.0, 1.0]]), 17.5393),
```
Comment (MalteEbner) [inline tests/loss/test_koleo_loss.py:17]: These two don't make sense, as they only calculate `-ln(eps + float_inaccuracies)`. I would only keep one of them, to ensure that a batch size of 1 does not break.

However, can you add these?
- [[0.0], [1.0]] -> should give mean([-ln(1), -ln(1)])
- [[0.0], [1.0], [2.0] -> should give mean([-ln(1), -ln(2)])

**Review by MalteEbner — COMMENTED** [review]
Can you add a bit more tests? Ideally, they should not only test equality with the original implementation, but also against known easy values.

Otherwise looks good.

File: tests/loss/test_koleo_loss.py (hunk 1 above)
Comment (guarin) [reply tests/loss/test_koleo_loss.py]: Added more tests :)

Note that tests like this don't really make sense: `[[0.0], [1.0], [2.0]]`. The reason is that the loss first normalizes the vectors (`[0.0]`, `[1.0]`, `[2.0]`) to unit norm. So you'll always end up with a `[1.0]` vector (except for 0.0 which remains 0.0).

**Review by MalteEbner — APPROVED** [review]
